eng/pipelines/libraries/build-job.yml

```diff
@@ -81,27 +81,102 @@ jobs:
         - ${{ if eq(parameters.useHelix, true) }}:
           - _additionalBuildArguments: /p:ArchiveTests=true
 
+        # Currently only support pipeline caching for PRs and manual triggered runs
+        - ${{ if or(in(variables['Build.Reason'], 'PullRequest'), in(variables['Build.Reason'], 'Manual')) }}:
+          - name: UsePipelineCache
+            value: true
+
+        - name: ConfigurationName
+          value: ${{ format('Libraries {0} {1}{2} {3} {4} {5}', parameters.runtimeVariant, parameters.osGroup, parameters.osSubgroup, parameters.archType, parameters.buildConfig, parameters.framework) }}
+
+        - name: CommitsOutputFile
+          value: '$(Build.SourcesDirectory)/libraries.commit_hash'
+
+        - name: GetCommitScript
+          value: '$(Build.SourcesDirectory)/src/coreclr/scripts/get_commit_hash.py'
+
+        # Otherwise set all the cache-specific variables to 'false' which will ensure to build product.
+        - ${{ if and(notin(variables['Build.Reason'], 'PullRequest'), notin(variables['Build.Reason'], 'Manual')) }}:
+          - name: UsePipelineCache
+            value: false
+          - name: BUILD_CACHE_RESTORED
+            value: false
+          - name: BUILD_LOGS_RESTORED
+            value: false
+
+        - ${{ if eq(parameters.osGroup, 'windows') }}:
+          - name: PythonScript
+            value: 'py -3'
+          - name: PipScript
+            value: 'py -3 -m pip'
+
+        - ${{ if ne(parameters.osGroup, 'windows') }}:
+          - name: PythonScript
+            value: 'python3'
+          - name: PipScript
+            value: 'pip3'
+
         - ${{ parameters.variables }}
 
       steps:
+
+        # Get the commit hashes of interesting folder for caching
```

Comment:
Should we put all these common cache steps on a common template file and then configure it via parameters?